src/rand.jl

```diff
@@ -0,0 +1,58 @@
+import Random
+
+function Random.Sampler(
+    RNG::Type{<:Random.AbstractRNG},
+    st::Random.SamplerType{TOrRef},
+    n::Random.Repetition,
+) where {TOrRef<:Union{Arf,ArfRef,Arb,ArbRef,Acb,AcbRef}}
+    T = _nonreftype(TOrRef)
+    # return Random.SamplerSimple(
+    #     Random.SamplerType{T}(),
+    #     Random.SamplerBigFloat(precision(T)),
+    # )
+    res = setprecision(BigFloat, precision(T)) do
+        Random.SamplerSimple(Random.SamplerType{T}(), Random.Sampler(RNG, BigFloat, n))
+    end
+    return res
+end
+
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Arf}) where {A,B} =
+    Arf(rand(rng, sp.data), prec = sp.data.prec)
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Arb}) where {A,B} =
+    Arb(rand(rng, sp.data), prec = sp.data.prec)
+Random.rand(rng::Random.AbstractRNG, sp::Random.SamplerSimple{A,B,Acb}) where {A,B} =
+    Acb(rand(rng, sp.data), rand(rng, sp.data), prec = sp.data.prec)
+
+function Random.rand(
+    rng::Random.AbstractRNG,
+    st::Random.SamplerTrivial{A,A},
+) where {A<:ArfOrRef}
```

Comment:
Maybe naive question, but why define `rand` for "scalars" here? Is it like calling `rand(big(1.3))`  and getting a random `BigFloat` of the same precision as the input?

Reply:
yes, if we already have `x = Arb("1.1", prec=128)`, I'd like to have `precision(rand(x)) == precision(x)`

Reply:
Ok; I just find it weird, as `rand(x)` doesn't depend on the value of `x`, just on its precision. But alternatives I can think of are obviously more verbose :shrug:

Reply:
indeed, but since we're not carrying precision in the type, that was the easiest way I could thought how to pass it.
I think this is useful in iterative algorithms where you double precision of computations every step, and you need randomness that stretches far outside Float64 range